experimental/devices/cap1188/cap1188.go

```diff
@@ -0,0 +1,550 @@
+
+// The device is a 8 Channel Capacitive Touch Sensor with 8 LED Drivers
+//
+// Datasheet
+//
+// The official data sheet can be found here:
+//
+// http://ww1.microchip.com/downloads/en/DeviceDoc/CAP1188.pdf
+//
+package cap1188
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"log"
+	"time"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/gpio"
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/mmr"
+	"periph.io/x/periph/conn/spi"
+	"periph.io/x/periph/devices"
+)
+
+// TouchStatus is the status of an input sensor
+type TouchStatus int8
+
+func (t TouchStatus) String() string {
+	switch t {
+	case OffStatus:
+		return "Off"
+	case PressedStatus:
+		return "Pressed"
+	case HeldStatus:
+		return "Held"
+	case ReleasedStatus:
+		return "Released"
+	default:
+		return "Unknown"
+	}
+}
+
+const (
+	// OffStatus indicates that the input sensor isn't being activated
+	OffStatus TouchStatus = iota
+	// PressedStatus indicates that the input sensor is currently pressed
+	PressedStatus
+	// HeldStatus indicates that the input sensor was pressed and is still held pressed
+	HeldStatus
+	// ReleasedStatus indicates that the input sensor was pressed and is being released
+	ReleasedStatus
+)
+
+const (
+	nbrOfLEDs = 8
+)
+
+const (
+	// reg_LEDLinking - The Sensor Input LED Linking register controls whether a
+	// capacitive touch sensor input is linked to an LED output. If the
+	// corresponding bit is set, then the appropriate LED output will change
+	// states defined by the LED Behavior controls in response to the capacitive
+	// touch sensor input.
+	reg_LEDLinking = 0x72
+	// reg_LEDOutputControl - The LED Output Control Register controls the output
+	// state of the LED pins that are not linked to sensor inputs.
+	reg_LEDOutputControl = 0x74
+)
+
+// Dev is a handle to a cap1188.
+type Dev struct {
+	// Debug is a flag that indicates that more logging will be created
+	Debug         bool
+	d             conn.Conn
+	regWrapper    mmr.Dev8
+	isSPI         bool
+	opts          *Opts
+	inputStatuses []TouchStatus
+}
+
+func (d *Dev) String() string {
+	return fmt.Sprintf("cap1188{%s}", d.regWrapper.Conn)
+}
+
+// Halt is a noop for the cap1188.
+func (d *Dev) Halt() error {
+	return nil
+}
+
+// InputStatus reads and returns the status of the 8 inputs as an array where
+// each entry indicates a touch event or not.
+func (d *Dev) InputStatus() ([]TouchStatus, error) {
+
+	// read inputs
+	status, err := d.regWrapper.ReadUint8(0x3)
+	if err != nil {
+		return d.inputStatuses, wrap(fmt.Errorf("failed to read the input values - %s", err))
+	}
+
+	// read deltas (in two's complement, capped at -128 to 127)
+	deltasB := [nbrOfLEDs]byte{}
+	if err = d.regWrapper.ReadStruct(0x10, &deltasB); err != nil {
+		return d.inputStatuses, wrap(fmt.Errorf("failed to read the delta values - %s", err))
+	}
+	// twoComplementsToInt returns the int value of a standard 2’s complement number
+	twoComplementsToInt := func(v byte) int {
```

Comment:
I mean to inline the code directly below, e.g.

```go
for i, b := range deltasB {
  if b&0x80 == 0 {
    deltas[i] = int(b)
  } else {
    deltas[i] = int(b) - 0x100
  }
 }
```

but I think just doing this would work:
```go
for i, b := range deltasB {
  deltas[i] = int(int8(b))
 }
```